File Dropzone: Add accept property

ElishaSamPeterPrabhu opened the pull request:
## Description

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Added accept property to accept certain file types

References 
Fixes #2003 <!-- issue number -->

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

coliff commented:
Quick few thoughts on this:
- The property is labeled `accept` which I don't think is descriptive enough. Perhaps `accept-file-types` would be better?
- In the docs/canvas demo it would be helpful to have an example of how exactly a file type to be added. What format does it accept them? would `accept=".doc,.docx"` work like [input type=file](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file) does?

ElishaSamPeterPrabhu commented:
> * .doc,.docx



> Quick few thoughts on this:
> 
> * The property is labeled `accept` which I don't think is descriptive enough. Perhaps `accept-file-types` would be better?
> * In the docs/canvas demo it would be helpful to have an example of how exactly a file type to be added. What format does it accept them? would `accept=".doc,.docx"` work like [input type=file](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file) does?

- Renamed the prop to `acceptFileTypes` and added formats doc to the story book
- While adding more than one format it accepts files in this way
 
![image](https://github.com/trimble-oss/modus-web-components/assets/122255626/cb36fb88-4d4a-4a10-aa43-121697053411)

- For doc alone its accepts files in this way
![image](https://github.com/trimble-oss/modus-web-components/assets/122255626/9d78f6bd-c1c4-42f2-abae-56c42dd3668d)
